### PR TITLE
current logged in user api

### PR DESCRIPTION
**Pull request by ntxnick**
### Pre-Requisites

- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevant code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

### Description of Changes

Adds API to check the currently logged-in user.

### Related Issues

Closes #74

### CODE REVIEW

**Comment by SpencerKaiser** [comment]
This looks like it shares a [bunch of code](https://github.com/AmericanAirlines/Project-X/pull/62/files#diff-ca9fa2ba51ed407a6e139651fdabd1c22906b0aa1e722c738a1d017599e0dad1R18-R32) from #62... marking this as draft until we get more info

**Review by dre8597 — APPROVED** [review]
LGTM  🚀

File: packages/api/tests/api/users.test.ts [hunk 1]
```diff
@@ -12,9 +12,82 @@ const sampleUser: Partial<User> = {
...
+
 const loggerSpy = jest.spyOn(logger, 'error').mockImplementation();
 
-describe('users API GET route', () => {
+describe('GET users/me', () => {
+  beforeEach(async () => {
+    jest.clearAllMocks();
+  });
+
+  it('401 error when not logged in', async () => {
```
Comment (SpencerKaiser) [inline packages/api/tests/api/users.test.ts:29]: These should always read as a sentence starting with `it`
```suggestion
  it('returns a 401 error when not logged in', async () => {
```

File: packages/api/tests/api/users.test.ts [hunk 2]
```diff
@@ -12,9 +12,82 @@ const sampleUser: Partial<User> = {
...
+    const handler = testHandler(users);
+
+    const { text } = await handler.get('/me').expect(401);
+    expect(text).toEqual('You must be logged in.');
+    expect(handler.entityManager.findOne).toBeCalledTimes(0);
+  });
+
+  it('404 error current user not found in database', async () => {
```
Comment (SpencerKaiser) [inline packages/api/tests/api/users.test.ts:37]:
```suggestion
  it('returns a 404 error when the current user is not found', async () => {
```

File: packages/api/tests/api/users.test.ts [hunk 3]
```diff
@@ -12,9 +12,82 @@ const sampleUser: Partial<User> = {
...
+    handler.entityManager.findOne.mockResolvedValueOnce(null);
+
+    await handler.get('/me').expect(404);
+    expect(handler.entityManager.findOne).toBeCalledTimes(1);
+    expect(handler.entityManager.findOne).toHaveBeenCalledWith(User, {
+      githubId: sampleSignedInUser.profile.id,
+    });
+  });
+
+  it('500 error during findOne', async () => {
```
Comment (SpencerKaiser) [inline packages/api/tests/api/users.test.ts:52]:
```suggestion
  it('returns a 500 error when a database query fails', async () => {
```

File: packages/api/tests/api/users.test.ts [hunk 4]
```diff
@@ -12,9 +12,82 @@ const sampleUser: Partial<User> = {
...
+
+    expect(loggerSpy).toBeCalledTimes(1);
+    expect(handler.entityManager.findOne).toBeCalledTimes(1);
+    expect(handler.entityManager.findOne).toHaveBeenCalledWith(User, {
+      githubId: sampleSignedInUser.profile.id,
+    });
+    expect(text).toEqual('There was an issue getting the currently logged in user');
+  });
+
+  it("Successfully send current user's information", async () => {
```
Comment (SpencerKaiser) [inline packages/api/tests/api/users.test.ts:70]: Also try not to use apostrophes in the description; it makes Jest very, _**very**_ unhappy sometimes
```suggestion
  it("successfully sends the data for the user when the user is logged in", async () => {
```

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]
Quick changes and then we can merge!

**Review by SpencerKaiser — APPROVED** [review]

**Pull request merged**